File: IM.eSuite.Domain/Model/UserPlant.cs
using IM.NETCore.Domain;

namespace IM.eSuite.Domain
{
    public class UserPlant
    {
        public int UserId { get; set; }
        public int PlantId { get; set; }

        // Navigation properties
        public virtual User User { get; set; }
        public virtual Plant Plant { get; set; }
    }
}

File: IM.eSuite.Web/src/app/modules/administration/models/userPlant.ts
import { Plant } from "../../organization/models/plant";

export class UserPlant {
    userId: number;
    plantId: number;
    plant: Plant;

    constructor(userPlant?: UserPlant) {       
        
        if(userPlant){
            this.userId = userPlant.userId;
            this.plantId = userPlant.plantId;
            this.plant = new Plant(userPlant.plant);
        }
        else {
            this.plant = new Plant();
        }
    }
}